coremltools/converters/mil/frontend/torch/test/testing_utils.py

```diff
@@ -249,16 +249,21 @@ def run_compare_torch(
         if minimum_deployment_target is not None:
             validate_minimum_deployment_target(minimum_deployment_target, backend)
 
-        model.eval()
         if input_as_shape:
             input_data = generate_input_data(input_data, rand_range, torch_device)
 
         if frontend == TorchFrontend.TORCHSCRIPT:
+            model.eval()
             if use_scripting:
                 model_spec = torch.jit.script(model)
             else:
                 model_spec = trace_model(model, _copy_input_data(input_data))
         elif frontend == TorchFrontend.EXIR:
+            try:
+                model.eval()
+            except NotImplementedError:
```

Comment:
Maybe raise a warning message here?
If the model is not in eval mode,
are we expecting seeing some strange downstream error?

Reply:
Raised warning message as suggested.

I guess we wouldn't, since `torch.export` should already be targeting inference, i.e. all EXIR program should already be in "eval" mode